samples/packages/asset-import/src/test/kotlin/com/atlan/pkg/aim/DeletedLinkTest.kt

```diff
@@ -0,0 +1,167 @@
+        val db = Database.creator(DB_NAME, connection.qualifiedName).build()
+        val response = db.save(client)
+        return response.getResult(db)
+    }
+
+    private fun createLink(database: Database): String {
+        val l =
+            Link
+                .creator(
+                    database.trimToReference(),
+                    LINK_NAME,
+                    LINK_URL,
+                ).build()
+        var response = l.save(client)
+        val link = response.getResult(l)
+        val linkGuid = link.guid
+        response = Asset.delete(client, linkGuid)
+        assertEquals(response.deletedAssets.size, 1)
+        return linkGuid
+    }
+
+    override fun setup() {
+        connection = createConnection()
+        linkGuid = createLink(createDatabase(connection))
+        prepFile()
+        runCustomPackage(
+            AssetImportCfg(
+                assetsFile = Paths.get(testDirectory, testFile).toString(),
+                assetsUpsertSemantic = "upsert",
+                assetsFailOnErrors = true,
+                assetsDeltaSemantic = "full",
+            ),
+            Importer::main,
+        )
+    }
+
+    override fun teardown() {
+        removeLink()
+        removeConnection(conn1, conn1Type)
+    }
+
+    private fun removeLink() {
+        Asset.purge(client, linkGuid) //
+    }
+
+    @Test(groups = ["aim.ctud.create"])
+    fun connection1Created() {
+        validateConnection()
+    }
+
+    private fun validateConnection() {
+        val found = Connection.findByName(client, conn1, conn1Type, connectionAttrs)
+        assertNotNull(found)
+        assertEquals(1, found.size)
+        val c1 = found[0]
+        assertEquals(conn1, c1.name)
+        assertEquals(conn1Type, c1.connectorType)
+    }
+
+    @Test(groups = ["aim.ctud.create"])
+    fun database1Created() {
+        validateLinkPresent()
+    }
+
+    private fun validateLinkPresent() {
+        val c1 = Connection.findByName(client, conn1, conn1Type, connectionAttrs)[0]!!
+        val request =
+            Database
+                .select(client)
+                .where(Database.CONNECTION_QUALIFIED_NAME.eq(c1.qualifiedName))
+                .includesOnResults(databaseAttrs)
+                .includeOnRelations(Schema.NAME)
+                .includeOnRelations(Link.NAME)
+                .includeOnRelations(Link.LINK)
+                .includeOnRelations(Link.STATUS)
+                .toRequest()
+        val response = retrySearchUntil(request, 1)
+        val found = response.assets
+        assertEquals(1, found.size)
+        val db = found[0] as Database
+        assertEquals(DB_NAME, db.name)
+        assertEquals(c1.qualifiedName, db.connectionQualifiedName)
+        assertEquals(conn1Type, db.connectorType)
+        val links = db.links
+        assertEquals(links.size, 1)
+        val link = links.elementAt(0)
+        assertEquals(link.name, LINK_NAME)
+        assertEquals(link.link, LINK_URL)
```

Comment:
We probably need to confirm that the link itself is also an active asset, don't we? (Could be that it's being returned as related, but actually the link itself is marked as deleted?)